src/mirakuru/base.py

```diff
@@ -129,12 +129,15 @@ def __init__(  # pylint:disable=too-many-arguments
 
         """
         if isinstance(command, (list, tuple)):
-            self.command = ' '.join(command)
+            full_command = ' '.join(command)
             """Command that the executor runs."""
-            self.command_parts = command
+            command_parts = command
         else:
-            self.command = command
-            self.command_parts = shlex.split(command)
+            full_command = command
+            command_parts = shlex.split(command)
+
+        self.command = shlex.quote(full_command)
```

Comment:
So if `command == ["a b", "c d"]`, then we get `full_command == "a b c d"` and `self.command == "'a b c d'"` which is incorrect.